File: atcoder/abc393b_a_b_c.py
# Vicfred
# https://atcoder.jp/contests/abc393/tasks/abc393_a
# implementation

def count_abc_triplets(S: str) -> int:
    n = len(S)
    triples = 0
    for i in range(n):
        for j in range(i + 1, n):
            for k in range(j + 1, n):
                if j - i != k - j:
                    continue
                if S[i] == 'A' and S[j] == 'B' and S[k] == 'C':
                    triples += 1
    return triples

def main():
    # Read input string
    S = input().strip()
    print(count_abc_triplets(S))

if __name__ == "__main__":
    main()

File: atcoder/abc393b_a_b_c.nim
# Vicfred
# https://atcoder.jp/contests/abc393/tasks/abc393_a
# implementation

import strutils

proc countABCTriplets(s: string): int =
    var triples = 0
    let n = s.len
    
    for i in 0..<n:
        for j in i+1..<n:
            for k in j+1..<n:
                if j - i != k - j:
                    continue
                if s[i] == 'A' and s[j] == 'B' and s[k] == 'C':
                    inc(triples)
    return triples

when isMainModule:
    # Read input string
    var s: string
    if stdin.readLine(s):
        s = s.strip()
        echo countABCTriplets(s)
    else:
        echo "Invalid input"
